src/java/org/apache/cassandra/net/MessageDelivery.java

```diff
@@ -74,9 +85,162 @@ public void onFailure(InetAddressAndPort from, RequestFailureReason reason)
     public <REQ, RSP> void sendWithCallback(Message<REQ> message, InetAddressAndPort to, RequestCallback<RSP> cb);
     public <REQ, RSP> void sendWithCallback(Message<REQ> message, InetAddressAndPort to, RequestCallback<RSP> cb, ConnectionType specifyConnection);
     public <REQ, RSP> Future<Message<RSP>> sendWithResult(Message<REQ> message, InetAddressAndPort to);
+
+    public default <REQ, RSP> Future<Message<RSP>> sendWithRetries(Backoff backoff,
+                                                                   RetryScheduler retryThreads,
+                                                                   Verb verb, REQ request,
+                                                                   InetAddressAndPort candidate,
+                                                                   TriFunction<Integer, InetAddressAndPort, RequestFailureReason, Boolean> shouldRetry,
```

Comment:
Should we make a special interface for this for convenience?

Reply:
sure, went with `MessageDelivery.RetryPredicate`